app/assets/javascripts/discourse/app/components/group-manage-email-settings.js

```diff
@@ -0,0 +1,117 @@
+import Component from "@ember/component";
+import discourseComputed from "discourse-common/utils/decorators";
+import I18n from "I18n";
+import bootbox from "bootbox";
+import { action } from "@ember/object";
+
+export default Component.extend({
+  tagName: "",
+
+  group: null,
+  clearImapEmailSettingsOnSave: false,
+  clearAllEmailSettingsOnSave: false,
+
+  @discourseComputed(
+    "group.emailSettingsValid",
+    "group.smtp_enabled",
+    "group.imap_enabled"
+  )
+  enableImapSettings(emailSettingsValid, smtpEnabled, imapEnabled) {
+    return smtpEnabled && (emailSettingsValid || imapEnabled);
+  },
+
+  @discourseComputed(
+    "group.emailSettingsValid",
+    "clearImapEmailSettingsOnSave",
+    "clearAllEmailSettingsOnSave"
+  )
+  disableSaveButton(
+    emailSettingsValid,
+    clearImapEmailSettingsOnSave,
+    clearAllEmailSettingsOnSave
+  ) {
+    return (
+      !emailSettingsValid &&
+      !clearImapEmailSettingsOnSave &&
+      !clearAllEmailSettingsOnSave
+    );
+  },
+
+  @action
+  smtpEnabledChange(event) {
+    if (!event.target.checked && this.group.smtp_enabled) {
+      bootbox.confirm(
+        I18n.t("groups.manage.email.smtp_disable_confirm"),
+        (result) => {
+          this.set("clearAllEmailSettingsOnSave", result);
+
+          if (!result) {
+            this.group.set("smtp_enabled", true);
+          } else {
+            this.group.set("imap_enabled", false);
+          }
+        }
+      );
+    }
+
+    if (event.target.checked) {
+      this.set("clearAllEmailSettingsOnSave", false);
+    }
+
+    this.group.set("smtp_enabled", event.target.checked);
```

Comment:
Are you talking about this with the buffered proxy? Seems like we only use it in one place, and I am not sure what the benefit is over using just a simple object to shadow these properties?

https://github.com/discourse/discourse/blob/da9051ecf12598396fcbd18ec4511cbf08e1f432/app/assets/javascripts/discourse/app/controllers/reorder-categories.js#L17-L22

Reply:
We use it in [many places](https://github.com/discourse/discourse/search?q=bufferedProperty) via `bufferedProperty`. One recent place I used is in the [`create-invite` modal](https://github.com/discourse/discourse/blob/60be1556fc3d25231323b47125c2b3ab67031788/app/assets/javascripts/discourse/app/controllers/create-invite.js) where the use case is similar.

The biggest benefit I find is that you can achieve more with less code, but I guess there are other benefits such as the initial object never being "dirty" - in most cases it will be an exact copy of what is on the server and that you do not have to copy the value of the properties and keep them up-to-date when they are updated in the original object.